driver/sources/src/main/scala/org/virtuslab/ideprobe/ClassFixtures.scala

```diff
@@ -22,33 +25,43 @@ trait RunningIntelliJPerSuiteBase {
   protected def afterAll(): Unit = ()
 
   def setup(): Unit = {
-    workspace = baseFixture.setupWorkspace()
-    installed = baseFixture.installIntelliJ()
-    running = baseFixture.startIntelliJ(workspace, installed)
-    runningIntelliJFixture = new RunningIntelliJFixture(workspace, running.probe, baseFixture.config, installed.paths)
+    workspace = Some(baseFixture.setupWorkspace())
```

Comment:
Maybe also imperfect, but I would structure this method differently. Options and handling options should not be visible throughout this method. Current implementation below this comment suggest that `workspace`, `installed` or anything can be null/None and it needs to be accessed through for comprehension. But actually it doesn't, none of the setup methods return option, and they don't return null (and they really, really shouldn't). I would separate **initialization code** (the original code) from assigning fields to what they expect.

```scala
    val workspace = baseFixture.setupWorkspace()
    this.workspace = Some(workspace)
    val installed = baseFixture.installIntelliJ()
    this.installed = Some(installed)
    val running = baseFixture.startIntelliJ(workspace, installed)
    this.running = Some(running)
    this.runningIntelliJFixture = Some(new RunningIntelliJFixture(workspace, running.probe, baseFixture.config, installed.paths))
    beforeAll()
```

It is not that nice because of assignment and logic are mixed with each other, but it is necessary in case setup fails at any step, so that it will be cleaned up. But hopefully it is more clear.

Another alternative would be to leave fields as they were and leave this method as it was, just in the teardown do the nullchecks in form of `Option(running).foreach(baseFixture.closeIntellij)`. I think either options is fine, all are imperfect, but I think more readable than current.